text/0000-fabric-gateway.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: default
+title: Fabric Gateway
+nav_order: 3
+---
+
+- Feature Name: Fabric Gateway
+- Start Date: 2020-09-08
+- RFC PR: (leave this empty)
+- Fabric Component: fabric-gateway
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The Fabric Gateway is a new component that will run either as a standalone process, or embedded with the peer, and will implement as much of the high-level 'gateway' programming model as possible.  This will remove much of the transaction submission logic from the client application which will simplify the maintenance of the SDKs.  It will also simplify the administrative overhead of running a Fabric network since the client application will only need to connect to a gateway rather than multiple peers across potentially multiple organisations.
+
+# Motivation
+[motivation]: #motivation
+
+The high-level "gateway" programming model has proved to be very popular with users since its introduction in v1.4 in Node SDK.  Since then, it has been implemented in the Java and Go SDKs.  This has led to a large amount of code that has to be maintained in the SDKs which increases the cost of implementing new feature which have to be replicated to keep the SDKs in step.  It also increases the cost of creating an SDK for a new language.  Moving all of the high-level logic into its own process and exposing a simplified GRPC interface will drastically reduce the complexity of the SDKs.
+
+By providing a single entry point to a Fabric network, client applications can interact with complex network topologies without multiple ports having to be opened and secured to each peer and ordering node.  The client will connect only to the gateway which will act on its behalf with the rest of the network.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The Fabric Gateway is an embodiment of the high-level 'gateway' Fabric programming model in a server component that will form part of a Fabric network alongside Peers, Orderers and CAs.  It will either be stood up inside its own process in a docker container, or it will be hosted inside a peer.  Either way, it exposes its functionality to clients via a GRPC interface.
```

Comment:
Given it exposes functionality via gRPC, i guess it'll be implemented once, in Go?

Reply:
Yes, one implementation in Go. Client only interacts using gRPC so doesn't care about the Fabric Gateway implementation. Similar to the existing nodes.

Reply:
Why not have a single implementation?

Reply:
I just said it was a single implementation. I don't understand the question.

Reply:
misread, I thought you said "one of them is in go"

Reply:
A assumption here is grpc is easy to learn to use and have been widely adopted. 
Per my observation it is not realistic. Developer still has to do restful-grpc translation to interact with off chain partners. 

Nowadays, application based on gateway has also incorporate this translator in a rich middleware layer. If this RFC provide a new process in a docker container, then they either have to add another layer or choose not to use.